### PR TITLE
Unit test for discovery panel

### PR DESCRIPTION
**Pull request by mistermendez**

### CODE REVIEW

File: js/components/discoveryPanel.js [hunk 1]
```diff
@@ -28,9 +26,39 @@ var DiscoveryPanel = React.createClass({
         })
       }),
       icons: React.PropTypes.objectOf(React.PropTypes.object)
+    }),
```
Comment (chibzu) [inline js/components/discoveryPanel.js]: Why use the shape type here?

File: js/components/discoveryPanel.js [hunk 2]
```diff
@@ -28,9 +26,39 @@ var DiscoveryPanel = React.createClass({
...
+    controller: React.PropTypes.shape({
+      sendDiscoveryClickEvent: React.PropTypes.func
     })
   },
 
+  getDefaultProps: function () {
```
Comment (chibzu) [inline js/components/discoveryPanel.js]: Any benefit here outside of testing? Could we mock this up in the tests instead?

Comment (mistermendez) [inline js/components/discoveryPanel.js]: make the component less fragile by ensuring props have values that may not have been specified by parent component.

better summary [here](https://facebook.github.io/react/docs/reusable-components.html#default-prop-values)

File: js/components/discoveryPanel.js (hunk 1 above)
Comment (mistermendez) [inline js/components/discoveryPanel.js]: I need to drill in and explicitly define proptype for sendDiscoveryClickEvent.

The controller proptype is less important.

Comment (chibzu) [inline js/components/discoveryPanel.js]: Let's change the controller type to be object to avoid confusion with external developers.

Comment (mistermendez) [inline js/components/discoveryPanel.js]: donezo

**Pull request merged**